tokio-trace-subscriber/src/lib.rs

```diff
@@ -18,10 +18,14 @@ pub use observe::ObserveExt;
 ///
 /// Implementations of this trait describe the logic needed to process envent
 /// and span notifications, but don't implement span registration.
-pub trait Observe: Filter {
+pub trait Observe {
     fn observe_event<'event, 'meta: 'event>(&self, event: &'event Event<'event, 'meta>);
     fn enter(&self, span: &SpanData);
     fn exit(&self, span: &SpanData);
+
+    fn filter(&self) -> &dyn Filter {
```

Comment:
I considered making the `&dyn Filter` return type an associated type instead (`&Self::Filter`), to avoid potential trait object stickiness later...but that then means that the default impl of no filter requires associated type defaults, which is currently unstable (rust-lang/rust#29661). Given that most of the motivation for changing the observer-filter relationship from is-a to has-a was that it would allow a default filter, I stuck with the trait object reference.